test/api/oran_o2ims_suite_test.go

```diff
@@ -0,0 +1,324 @@
+package main_test
+
+import (
+	"bytes"
+	"os"
+	"testing"
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestOranO2ims(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "OranO2ims Test Suite")
+}
+
+var _ = BeforeSuite(func() {
+	Expect(os.Getenv("TEST_HOST")).NotTo(BeZero(), "Please make sure TEST_HOST is set correctly.")
+})
+
+var _ = Describe("Metadata Server API testing", func() {
+	Context("When getting infrastructure Inventory API version", func() {
+		It("should return OK in the response and json response should match reference json", func() {
+			requestBody := []byte(`{"uriPrefix":"/o2ims-infrastructureInventory/v1","apiVersions":[{"version":"1.0.0"}]}`)
+			request, _ := http.NewRequest("GET", "https://"+os.Getenv("TEST_HOST")+"/o2ims-infrastructureInventory/api_versions", bytes.NewBuffer([]byte(requestBody)))
+			request.Header.Set("Content-Type", "application/json")
+
+			By("Executing https petition")
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			client := &http.Client{Transport: tr}
```

Comment:
The creation of the transport and client is repeated multiple times. You may want to use a `BeforeEach` block to write it only once:

```go
var _ = Describe("...", func() {
        var client *http.Client

        BeforeEach(func() {
                tr := &http.Transport{...}
                client = &http.Client{
                        Transport: tr
                }
        })
}
```

Then you can use the `client` variable in the tests.